### PR TITLE
Added support for 'softplus' and 'logSigmoid' and their VJPs.

### PR DESCRIPTION
**Pull request by eaplatanios**

### CODE REVIEW

File: Sources/TensorFlow/Layers/Upsampling.swift [hunk 1]
```diff
@@ -108,9 +108,6 @@ public struct UpSampling3D<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output.
     @differentiable
     public func call(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
-        let shape = input.shape
```
Comment (Shashi456) [inline Sources/TensorFlow/Layers/Upsampling.swift]: This seems to have been already deleted @eaplatanios

Comment (Shashi456) [reply Sources/TensorFlow/Layers/Upsampling.swift]: Oh this pops up in all the PRs, probably because of the matmul enhancement commit.

Comment (eaplatanios) [reply Sources/TensorFlow/Layers/Upsampling.swift]: Yes, that's right. We're going through the PRs in order and will merge with master before checking them in. :)

**Comment by rxwei** [comment]
Is this blocked on any other PR? If not, could you resolve the conflicts?

**Comment by eaplatanios** [comment]
I don't think it's blocked. I'll resolve the conflicts later today because I'm currently at a conference.

**Comment by eaplatanios** [comment]
@rxwei I finally updated this and ran tests locally. It should be ready to merge. The only failing tests I see have to do with the lazy tensor implementation and so are not relevant to this PR.

**Comment by eaplatanios** [comment]
Also verified that all CI tests pass.

File: Sources/TensorFlow/Operators/Math.swift [hunk 2]
```diff
@@ -1905,6 +1933,7 @@ internal func _vjpMatmul<Scalar: TensorFlowFloatingPoint>(
 ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> (Tensor<Scalar>, Tensor<Scalar>)) {
     let value = matmul(lhs, transposed: transposeA, rhs, transposed: transposeB)
     return (value, { v in
+<<<<<<< HEAD
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Math.swift]: Looks like merge conflicts were not resolved.

Comment (rxwei) [reply Sources/TensorFlow/Operators/Math.swift]: NVM!

File: Sources/TensorFlow/Operators/Math.swift [hunk 3]
```diff
@@ -923,6 +923,14 @@ internal func _vjpSigmoid<T: TensorFlowFloatingPoint>(
     (sigmoid(x), { v in Raw.sigmoidGrad(x, dy: v) })
 }
 
+/// Returns the log-sigmoid of the specified tensor element-wise. Specifically,
+/// `y = log(1 / (1 + exp(-x)))`. For numerical stability, we use `y = -softplus(-x)`.
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Math.swift]:
```suggestion
/// `log(1 / (1 + exp(-x)))`. For numerical stability, we use `-softplus(-x)`.
```

**Comment by eaplatanios** [comment]
@rxwei this was actually merged as part of #157. I realized I mixed up the PRs while trying to keep them updated in my local branches. I'll go through and see which ones remain now.

**Pull request closed without merging**